### PR TITLE
Builder Options for Entitlements, Grants, and Resources.

### PR DESCRIPTION
**Pull request by mstanbCO**
Adds options to the Entitlement, Grant, and Resources new object builders.

### CODE REVIEW

File: pkg/types/entitlement/entilement.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	}
+}
+
+func WithDisplayName(displayName string) EntitlementOption {
+	return func(g *v2.Entitlement) {
+		g.DisplayName = displayName
+	}
+}
+
+func WithDescriptionName(description string) EntitlementOption {
```
Comment (jirwin) [inline pkg/types/entitlement/entilement.go:33]: This should probably just be `WithDescription`

File: pkg/types/entitlement/entilement.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package entitlement
+
+import (
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"google.golang.org/protobuf/proto"
+)
+
+type EntitlementOption func(*v2.Entitlement)
```
Comment (jirwin) [inline pkg/types/entitlement/entilement.go:9]: What do you think of moving all of the `New*` functions into these smaller type specific packages so we wouldn't have any of this in the `sdk/` package?

Comment (mstanbCO) [reply pkg/types/entitlement/entilement.go:9]: Yeah, I think that makes more sense. I'll get a commit added.

File: pkg/types/resource/resource_test.go [hunk 3]
```diff
@@ -133,44 +63,28 @@ func TestNewGroupResource(t *testing.T) {
 	require.True(t, ok)
 	require.Equal(t, "v1", v1ID.Id)
 
-	roleTrait, err := GetRoleTrait(gr)
+	roleTrait, err := sdk.GetRoleTrait(gr)
```
Comment (jirwin) [inline pkg/types/resource/resource_test.go:66]: Probably better for another PR, but I think it makes sense to move these trait helpers to the types package as well. The `sdk` package doesn't make much sense I think, so moving more out of it feels good.

**Review by jirwin — APPROVED** [review]

**Pull request merged**